sktime/forecasting/statsforecast.py

```diff
@@ -552,3 +555,150 @@ def get_test_params(cls, parameter_set="default"):
         params = [{}, {"season_length": 4, "model": "Z"}]
 
         return params
+
+
+class StatsForecastMSTL(_GeneralisedStatsForecastAdapter):
+    """StatsForecast Multiple Seasonal-Trend decomposition using LOESS model.
+
+    This implementation is a wrapper over Nixtla implementation in
+    statsforecast [1]_.
+
+    The MSTL (Multiple Seasonal-Trend decomposition using LOESS) decomposes the time
+    series in multiple seasonalities using LOESS. Then forecasts the trend using
+    a custom non-seasonal model (`trend_forecaster`) and each seasonality using a
+    SeasonalNaive model. MSTL requires the input time series data to be univariate.
+
+    Parameters
+    ----------
+    season_length : Union[int, List[int]]
+        Number of observations per unit of time. For multiple seasonalities use a
+        list.
+    trend_forecaster : estimator, optional, default=StatsForecastAutoETS()
+        Sktime estimator used to make univariate forecasts. Multivariate estimators are
+        not supported.
+
+    References
+    ----------
+    .. [1]
+        https://nixtla.github.io/statsforecast/src/core/models.html#multiple-seasonalities
+
+    Examples
+    --------
+    >>> from sktime.datasets import load_airline
+    >>> from sktime.forecasting.statsforecast import StatsForecastMSTL
+
+    >>> y = load_airline()
+    >>> model = StatsForecastMSTL(season_length=[3,12]) # doctest: +SKIP
+    >>> fitted_model = model.fit(y=y) # doctest: +SKIP
+    >>> y_pred = fitted_model.predict(fh=[1,2,3]) # doctest: +SKIP
+    """
+
+    _tags = {
+        "ignores-exogeneous-X": True,
+        "capability:pred_int": True,
+        "capability:pred_int:insample": True,
+        "python_dependencies": ["statsmodels", "statsforecast"],
+    }
+
+    def __init__(
+        self,
+        season_length: Union[int, List[int]],
+        trend_forecaster=None,
+    ):
+        super().__init__()
+
+        if _check_soft_dependencies("statsmodels", severity="none"):
+            from statsforecast.models import _TS
+        else:
+            raise Exception(
+                "StatsForecastMSTL requires module `statsforecast` to be installed. "
+                "Please install `statsforecast` in your environment."
+            )
```

Comment:
I am not sure how `_TS` is being used here, but checking for `statsmodels` and importing `statsforecast` is not right.

Reply:
Also, one should never raise such a broad exception. If I'm not wrong, `_check_soft_dependencies` allow raising error itself.

Reply:
Typo on the `statsmodels` check
Also now using "error" `severity` of `_check_soft_dependencies`